index.js

```diff
@@ -6,10 +6,12 @@ function createFunction(parameters, defaultType) {
     if (!isFunctionDefinition(parameters)) {
         fun = function() { return parameters; };
         fun.isFeatureConstant = true;
-        fun.isGlobalConstant = true;
+        fun.isZoomConstant = true;
 
     } else {
-        var property = parameters.property === undefined ? '$zoom' : parameters.property;
+        var zoomAndDataDependent = typeof parameters.stops[0][0] === 'object';
+        var dataDependent = zoomAndDataDependent || parameters.property !== undefined;
```

Comment:
I'm not necessarily preferring "property functions" over "feature functions", that's just the terminology I've been test-driving lately. 

I've realized more recently that the `isPaintValueFeatureConstant` and `isPaintValueZoomConstant` would segue more nicely into the "feature functions" naming scheme.